api/admin/controller/collection_settings.py

```diff
@@ -0,0 +1,266 @@
+        # setting to every protocol's 'settings' block.
+        mirror_integration_setting = self._mirror_integration_setting()
+        if mirror_integration_setting:
+            for protocol in self.protocols:
+                protocol['settings'].append(mirror_integration_setting)
+
+    # GET
+    def process_get(self):
+        user = flask.request.admin
+        collections = []
+        protocolClass = None
+        for collection_object in self._db.query(Collection).order_by(Collection.name).all():
+            if not user or not user.can_see_collection(collection_object):
+                continue
+
+            collection_dict = self.collection_to_dict(collection_object)
+
+            if collection_object.protocol in [p.get("name") for p in self.protocols]:
+                [protocol] = [p for p in self.protocols if p.get("name") == collection_object.protocol]
+                self.load_libraries(collection_dict, collection_object, user, protocol)
+                self.load_settings(protocol.get("settings"), collection_dict, collection_object)
+                [protocolClass] = self.find_protocol_class(collection_object)
+
+            collection_dict["self_test_results"] = self._get_prior_test_results(collection_object, protocolClass)
+            collections.append(collection_dict)
+
+        return dict(
+            collections=collections,
+            protocols=self.protocols,
+        )
+
+    def collection_to_dict(self, collection_object):
+        return dict(
+            id=collection_object.id,
+            name=collection_object.name,
+            protocol=collection_object.protocol,
+            parent_id=collection_object.parent_id,
+            settings=dict(external_account_id=collection_object.external_account_id),
+        )
+
+    def load_libraries(self, collection_dict, collection_object, user, protocol):
+        libraries = []
+        for library in collection_object.libraries:
+            if not user or not user.is_librarian(library):
+                continue
+            libraries.append(self._get_integration_library_info(
+                    collection_object.external_integration, library, protocol))
+
+        collection_dict['libraries'] = libraries
+
+    def load_settings(self, protocol_settings, collection_dict, collection_object):
+        for protocol_setting in protocol_settings:
+            key = protocol_setting.get("key")
+            if key not in collection_dict.get("settings"):
+                if key == 'mirror_integration_id':
+                    value = collection_object.mirror_integration_id or self.NO_MIRROR_INTEGRATION
+                elif protocol_setting.get("type") == "list":
+                    value = collection_object.external_integration.setting(key).json_value
+                else:
+                    value = collection_object.external_integration.setting(key).value
+                collection_dict["settings"][key] = value
```

Comment:
Similarly for this one. It would be easier to understand this code if `load_settings` was responsible for creating the entire `settings` dictionary (including the `external_account_id=collection_object.external_account_id` bit which is currently defined in collection_to_dict), and then `process_get` assigned it to `collection_dict["settings"]`.